npiperelay.go

```diff
@@ -59,9 +97,73 @@ func main() {
 		log.Println("connecting to", args[0])
 	}
 
-	conn, err := dialPipe(args[0], *poll)
-	if err != nil {
-		log.Fatalln(err)
+	var conn *overlappedFile
+	var err error
+
+	// Loop only if we're polling the named pipe or socket
+	for {
+		conn, err = dialPipe(args[0], *poll)
+
+		if *poll && os.IsNotExist(err) {
+			time.Sleep(200 * time.Millisecond)
+			continue
+		}
+
+		if err != nil {
+			err = &os.PathError{Path: args[0], Op: "open", Err: err}
+			log.Fatalln(err)
+		}
+
+		// LibAssaaun file socket: Attempt to read contents of the target file and connect to a TCP port
+		if *assuan {
```

Comment:
I would prefer that the existing pipe path stay the same as it was (don't change `dialPipe`) and that this path be a separate branch, maybe a separate function `dialAssuan`.